Makefile

```diff
@@ -146,6 +146,11 @@ $(obj)/src/driver/linux_onload: $(obj)/src/lib/citools $(obj)/src/lib/ciul \
                                 $(obj)/src/lib/transport/ip \
                                 $(obj)/src/lib/cplane mkdirs
 
+# Linux-6.8 turns on -Wmissing-declarations and -Wmissing-prototypes globally.
+# Onload has too many places which trigger those warnings. So, just remove
+# the options to keep build working.
+KBUILD_CFLAGS := $(filter-out -Wmissing-prototypes -Wmissing-declarations, $(KBUILD_CFLAGS))
```

Comment:
I'm considering these (similar) changes for the new and old build systems:
```diff
diff --git a/Makefile b/Makefile
index fd610837b2..4e92360569 100644
--- a/Makefile
+++ b/Makefile
@@ -213,6 +213,9 @@ ifneq ($(MMAKE_LIBERAL),1)
 ONLOAD_CFLAGS += -Werror
 endif

+# FIXME Address these in the source code.
+ONLOAD_CFLAGS += -Wno-missing-prototypes -Wno-missing-declarations
+
 ONLOAD_MAKEFLAGS ?=

 ifeq ($(HAVE_SFC),1)
diff --git a/mk/linux_kbuild.mk b/mk/linux_kbuild.mk
index 4d617478a7..c842decfc0 100644
--- a/mk/linux_kbuild.mk
+++ b/mk/linux_kbuild.mk
@@ -30,6 +30,9 @@ endif
 ifndef MMAKE_LIBERAL
 EXTRA_CFLAGS += -Werror
 endif # MMAKE_LIBERAL
+#
+# FIXME Address these in the source code.
+EXTRA_CFLAGS += -Wno-missing-prototypes -Wno-missing-declarations

 ifdef W_NO_STRING_TRUNCATION
 EXTRA_CFLAGS += -Wno-stringop-truncation
```
As they seem less intrusive. Please let me know if it doesn't work for you.

Reply:
Looks good!

Reply:
Awesome. 👍 I'm happy to merge once the AF_XDP stuff is ready.